### ISSUE TITLE
quick start setup process can fail at `tdp-ui init`

### ISSUE THREAD

**Issue by mralbertk** [TOSIT-IO/tdp-getting-started#250]
The current version of the ui does not run `npm install` and also does not validate that dependencies (i.e. Java) are met. 

This can be a problem for new users who follow the [quick start](https://github.com/TOSIT-IO/tdp-getting-started#quick-start) because the UI is installed as part of the default command yet the set-up process breaks:

```sh
$ git clone https://github.com/TOSIT-IO/tdp-getting-started.git
...
$ cd tdp-getting-started 
$ ./scripts/setup.sh -e server -e ui -e extras -e observability -e prerequisites -e vagrant
...
tdp-ui init
npm ERR! code 127
npm ERR! path /home/albertk/repos/adaltas/tosit/tdp-getting-started/tdp-ui
npm ERR! command failed
npm ERR! command sh -c husky install
npm ERR! sh: 1: husky: not found

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/albertk/.npm/_logs/2023-02-24T09_57_41_256Z-debug-0.log
```

To solve locally (merci @PaulFarault): 

1) Run `npm install` in `tdp-getting-started/tdp-ui`
2) Re-run ` ./scripts/setup.sh -e server -e ui -e extras -e observability -e prerequisites -e vagrant`

That said, this is not a great experience for new users who want to try TDP.  I suggest to temporarily remove `-e ui` from the default setup command and instead provide it with the workaround in [Deploy with TDP UI](https://github.com/TOSIT-IO/tdp-getting-started#deploy-with-tdp-ui)

**Comment by rpignolet** [comment]
After reading `package.json` from `tdp-ui` I see this https://github.com/TOSIT-IO/tdp-ui/blob/6a6c42a409fed356c5fece868a7aaf1888ff5229/package.json#L11.

And I have `husky` in `node_modules` here `./node_modules/.bin/husky`.

I don't know why this is not working for you.

**Comment by mralbertk** [comment]
I did a new install on a clean installation of Ubuntu, so I had none of the prerequisites on my machine and installed all components for the first time.

Maybe @PaulFarault can jump in to clarify but my understanding is that the setup script for tdp-ui currently does not run `npm install`, hence the need to run it manually in order to install the packages.

**Comment by PaulFarault** [comment]
Your problem actually seems to be that `npm install` is not run. However, the `setup.sh` script seems to be fine:

https://github.com/TOSIT-IO/tdp-getting-started/blob/eab96b423b9aebd96e350b6bc061a2411fea6632/scripts/setup.sh#L304-L308

Are you sure you have done all prerequisites given in the [`tdp-ui` readme](https://github.com/TOSIT-IO/tdp-ui/blob/946490b09f615b2b27316082a7167aab8f24cb82/README.md?plain=1#L13-L18)?

I'll try to reproduce your issue. I'll keep you updated.

**Comment by mralbertk** [comment]
Hey @PaulFarault, I did not actively install any of the pre-requisites for `tdp-ui` since I was going to deploy with vagrant.

**Comment by PaulFarault** [comment]
Well, Vagrant and `tdp-ui` are different things. I recommend that you fulfill all the prerequisites if you plan to use the ui:

https://github.com/TOSIT-IO/tdp-getting-started/blob/eab96b423b9aebd96e350b6bc061a2411fea6632/README.md?plain=1#L22-L32

**Comment by PaulFarault** [comment]
Don't hesitate to reach us if you think that the README is not explicit enough. We could think about how to improve it.

**Comment by mralbertk** [comment]
> Well, Vagrant and `tdp-ui` are different things. I recommend that you fulfill all the prerequisites if you plan to use the ui:

The issue is that the current quick start fails at `tdp-ui init` for users who do not plan to use the UI, and who therefore do not check the UI prerequisites. 

So yes, the README should be more explicit here. Maybe consider adding the non-dev requirements for `tdp-ui` to the general requirements?

**Comment by PaulFarault** [comment]
At this point, tdp-ui is not stable enough for it to be interesting to copy its prerequisites.

It could be sth like :

> Options are provided to select what is installed by the `setup.sh`:
> 
> `-e ui` for [`tdp-ui`](https://github.com/TOSIT-IO/tdp-ui), see the UI's [prerequisites](https://github.com/TOSIT-IO/tdp-ui/blob/master/README.md#pre-requisites).
> ...

Any though @rpignolet

**Comment by rpignolet** [comment]
In the README, in the "Requirements" section, I read

> If you use TDP UI to deploy TDP see requirements in https://github.com/TOSIT-IO/tdp-ui.

In the "Quick Start" section I see the option for activating the UI with the setup script

> `-e ui`

Currently the "Quick Start" activate all features and this is not explicit. I will make an issue to explain how the setup scripts work and prerequisites for each feature before the "Quick Start".

**Comment by rpignolet** [comment]
I think the issue is resolved, the README change will be done in #252.
